### PR TITLE
Replace `jupyter-execute` with `plot` from docs

### PR DESCRIPTION
**Pull request by arnaucasau**
This PR replaces all the `.. jupyter-execute::` calls with `.. plot::`. The change will improve the docs in different ways. For example, it will allow us to leverage the `:alt:` option of the `plot` directive to set an alt text to all the images, which wasn't possible using the `jupyter-execute` directive, and, in addition, it will prevent us from different issues when building the documentation caused by the unreliable nature of `juptyer-execute` as explained in https://github.com/Qiskit/qiskit/pull/9346

In this PR in `qiskit/documentation` we can see the git diff caused by this replacement. The change is mainly the renaming of the images because some of them now have a dedicated code block: https://github.com/Qiskit/documentation/pull/2403/files

### CODE REVIEW

File: qiskit_ibm_runtime/fake_provider/__init__.py
Comment (arnaucasau) [inline qiskit_ibm_runtime/fake_provider/__init__.py:57]: This is necessary to allow the code to run in the same context as all previous plot directives. Otherwise, we would need to add all the imports and the relevant code again.

By adding `close-figs` to the directive we keep the context but close previous figures before running the code. Without this part, we would see repeated images all over the page.

For more information, see https://matplotlib.org/stable/api/sphinxext_plot_directive_api.html#options

File: qiskit_ibm_runtime/transpiler/passes/scheduling/__init__.py
Comment (arnaucasau) [inline qiskit_ibm_runtime/transpiler/passes/scheduling/__init__.py:344]: I divided the code into two blocks to display both circuits. Before, even though we ran `.draw()` twice, we were only showing the last circuit as we can see at the end of this page: https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.transpiler.passes.scheduling

File: qiskit_ibm_runtime/fake_provider/__init__.py
Comment (arnaucasau) [inline qiskit_ibm_runtime/fake_provider/__init__.py:34]: This plot showed three images and I splitting it into three `.. plot::` to be able to set an alt text to each image individually.

**Review by Eric-Arellano — APPROVED** [review]
Nice work! Please wait for a Runtime team member to approve before merging

**Pull request merged**
